lib/phoenix_live_view/channel.ex

```diff
@@ -110,6 +110,15 @@ defmodule Phoenix.LiveView.Channel do
         {:noreply, push_render(%{state | components: new_components}, diff, nil)}
 
       :noop ->
+        {module, id, _} = update
+
+        if function_exported?(module, :__info__, 1) do
+          # Only a warning, because there can be race conditions where a component is removed before a `send_update` happens.
+          IO.warn "send_update failed (there is no #{inspect(module)} with ID #{inspect(id)})", []
```

Comment:
Since this happens at runtime, let's use Logger! I also propose to make the error message a bit clearer:

```suggestion
          Logger.debug "send_update failed because component #{inspect(module)} with ID #{inspect(id)} does not exist or it has been removed"
```

WDYT?

Reply:
Note, you may need to require Logger explicitly before.  haven't tested this code. :)

Reply:
Love it!

Reply:
Currently trying to figure out how to test it. Logger is available but `ExUnit.CaptureLog.capture_log` isn't catching it…

Reply:
Not even with my suggestion to call `render` after sending the message?

Reply:
Nope, not even then. Tried throwing a `flush` in there, too. Doing:

```
test "warns if component doesn't exist", %{conn: conn} do
  {:ok, view, _html} = live(conn, "/components")

  assert ExUnit.CaptureLog.capture_log(fn ->
           send(view.pid, {:send_update, [{StatefulComponent, id: "nemo", name: "NEW-nemo"}]})
           render(view)
           Logger.flush()
           refute_receive {:updated, _}
  end) =~ "send_update failed because component Phoenix.LiveViewTest.StatefulComponent with ID \"nemo\" does not exist or it has been removed"
end
```

Getting:

```
Assertion with =~ failed
code:  assert ExUnit.CaptureLog.capture_log(fn ->
         send(view.pid, {:send_update, [{StatefulComponent, [id: "nemo", name: "NEW-nemo"]}]})
         render(view)
         Logger.flush()
         refute_receive({:updated, _})
       end) =~ "send_update failed because component Phoenix.LiveViewTest.StatefulComponent with ID \"nemo\" does not exist or it has been removed"
left:  ""
right: "send_update failed because component Phoenix.LiveViewTest.StatefulComponent with ID \"nemo\" does not exist or it has been removed"
stacktrace:
  test/phoenix_live_view/integrations/components_test.exs:174: (test)

```

Reply:
Ah, I do get an error log, so something about log levels. Digging further.

Reply:
My proposal: change Logger.debug to Logger.warn. Seems a reasonable level, and it will show up in tests.

Otherwise I think we'd either need to change the log level in config.exs to allow debug which has knock-on effects (lots of logging starts showing), or perhaps we could use the non-macro Logger.log, which seems a shame.